### PR TITLE
Reduce memory allocations in NamingStyle.Capitalize/DecapitalizeFirstLetter

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Less memory allocation. Just saw this at random.

### CODE REVIEW

File: src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs [hunk 1]
```diff
@@ -89,13 +89,23 @@ private IEnumerable<string> ApplyCapitalization(IEnumerable<string> words)
         private string CapitalizeFirstLetter(string word)
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs]: I should observe that I have no idea what this code does in the case of surrogate pairs.

**Comment by jasonmalinowski** [comment]
The integration run hit #20019.

File: src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs (hunk 1 above)
Comment (nil4) [reply src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs]: Would it make sense to test if the first char is already upper case, and just return the input string; i.e. bypassing `ToCharArray` and `new string(chars)` completely?

Comment (jasonmalinowski) [reply src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs]: Why yes, yes it would.

Comment (nil4) [reply src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs]: What I meant was something like this (avoid allocations if `word` does not need to change).

```csharp
private string CapitalizeFirstLetter(string word) {
  if (word == null || word.Length < 1 || char.IsUpper(word[0]))) return word;
  var chars = word.ToCharArray();
  chars[0] = char.ToUpper(chars[0]);
  return new string(chars);
}
```

Comment (jasonmalinowski) [reply src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs]: @nil4 Yep, understood perfectly from the start.

**Pull request merged**